### ISSUE TITLE
[🐛 Bug]: next-on-pages removes wasm imports from production build

### ISSUE THREAD

**Issue by blukai** [cloudflare/next-on-pages#494]
### next-on-pages environment related information

```console
System:
        Platform: linux
        Arch: x64
        Version: #1 SMP PREEMPT_DYNAMIC Sat Sep 23 22:53:27 UTC 2023
        CPU: (16) x64 AMD Ryzen 9 5900HS with Radeon Graphics
        Memory: 31 GB
        Shell: /usr/bin/zsh
Binaries:
        Node: 20.8.0
        Bun: N/A
        pnpm: N/A
        Yarn: N/A
        npm: 10.1.0
Package Manager Used: npm
Relevant Packages:
        @cloudflare/next-on-pages: 1.6.3
        vercel: N/A
        next: 13.5.4

```

### Description

when multiple files import wasm modules `next-on-pages` produces a broken production build.

```
A promise rejection was handled asynchronously. This warning occurs when attaching a catch handler to a promise after it rejected. (rejection #1)
ReferenceError: wasm_59cd556b704b5c8ea09030c5838471a4b2299032 is not defined
    at Object.Bf (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/webpack/e3d30189dcd18971da5d91ca8a0b3626.js:56:33396)
    at n (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:2157)
    at Object.Vf (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/webpack/e3d30189dcd18971da5d91ca8a0b3626.js:56:35967)
    at n (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:2157)
    at 8046 (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:4291)
    at n (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:2157)
    at file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:6080
    at o (file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:3549)
    at file:///tmp/tmp-79001-VJJb8KyehEsA/__next-on-pages-dist__/functions/bananas/opengraph-image.func.js:1:3805 {
  stack: ReferenceError: wasm_59cd556b704b5c8ea09030c583847…/functions/bananas/opengraph-image.func.js:1:3805,
  message: wasm_59cd556b704b5c8ea09030c5838471a4b2299032 is not defined
}
```

it seems like `next-on-pages` removes wasm imports from the generated output files.

### Reproduction

https://github.com/blukai/next-on-pages-wasm-import-bug

in the reproduction repo when both directories `src/app/hmm` and `src/app/banana` are present - production build is broken.

everything works fine if wasm module is imported only once within a code base (to prove that - in the reproduction repo remove `src/app/hmm`  directory -> build the project (`npm run build:next-on-pages`) -> launch it `wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat` -> go to http://localhost:8788/bananas/opengraph-image -> it works).

### Pages Deployment Method

None

### Would you like to help?

- [X] Would you like to help fixing this bug?

**Comment by dario-piotrowicz** [comment]
Hi @blukai thanks a lot for the issue 🙂 

I can't stress enough how great is that you've provided a minimal reproduction! thanks so very much it was of invaluable help! ❤️

I've fixed the issue in #494 if you want please give the [prerelease](https://github.com/cloudflare/next-on-pages/pull/495#issuecomment-1751802462) have a try, it should definitely fix the issue 🙂

___

PS: you ticked the checkbox mentioning that you'd like to help with this, I wanted to investigate the issue to give you guidance but then I've found the bug and was just simpler to fix it there and there instead, sorry about that 😓, but thanks so much for wanting to help! ❤️

**Comment by blukai** [comment]
thanks a lot for addressing this so promptly!
